app/AuthorityName.php

```diff
@@ -20,4 +18,14 @@ public function authority()
     {
         return $this->belongsTo(Authority::class);
     }
+
+    public function scopeWhereFirstNameLastName($query, $name)
+    {
+        $query->whereRaw(
+            'case when instr(name, ", ") then ' .
+            'concat(substring_index(name, ", ", -1), " ", substring_index(name, ", ", 1)) = ? ' .
+            'else name = ? end',
+            [$name, $name]
+        );
```

Comment:
Would this also work?


```suggestion
        $query
            ->where('name', $name)
            ->orWhereRaw(
                'concat(substring_index(name, ", ", -1), " ", substring_index(name, ", ", 1)) = ? ',
                $name
            );
```

Reply:
Sure, but I found it a bit ignorant, but I know that the usage of `case` is less readable...